### DISCUSSION TITLE
post/dapr-consul-name-resolution

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#13] [Announcements]
# post/dapr-consul-name-resolution

Mark Heath's Development Blog

https://markheath.net/post/dapr-consul-name-resolution

**Comment by white-label-development** [comment]
Thank you for this post - super helpful

**Comment by tzarot** [comment]
Hi Mark, can you please clarify where should I create the specific to the application <b>daprConfig.yaml</b> file, in the case of a FrontEnd .NET Web App?

**Reply by markheath** [reply]
If you're in self-hosted mode then it's just anywhere you can point at with the --config option. 
On Kubernetes you use kubectl apply to set a custom configuration: https://docs.dapr.io/operations/configuration/configuration-overview/

**Reply by tzarot** [reply]
Thank you so much. However, been in self-hosted mode, and launching sidecars from within docker-compose.yml i.e. => 
command: [ "./daprd", "-app-id", "myfrontend", "-app-port", "80"] 
When I added: ...,"-config", "../dapr/components/daprConfig.yaml" 
I end up with dapr sidecar not starting, due to cannot find file!

**Reply by tzarot** [reply]
Found the config issue, required absolute path: 
command: [ "./daprd", "-app-id", "mbackend", "-app-port", "80", "-config", "C:\\Users\\Tzarot\\.dapr\\components\\daprConfig.yaml" ]

However, running Consule and have done all steps in article still my front-end does not talk to the backend. The error is a Timeout...

**Reply by markheath** [reply]
I suspect the consul container would need to be started as part of your docker compose, and then you would access it with it's container name rather than localhost. I haven't tried using it from docker compose though.

**Comment by dmaxim** [comment]
Thanks Mark,  

I am evaluating Dapr for use with the build out of new services in my work environment.  The DNS issue for development on MacOS was going to be a deal breaker for me.  Thanks for the easy fix.

**Comment by Filiposu** [comment]
Mark you are my hero . I was stuck with issue and your post worked for me greatly. Dapr should make this article appear on their quickstart guide.

**Comment by mvacha** [comment]
Thank you for the only consul tutorial for dapr dummies on the internet.. I was stuck on weird issues w. mDNS on Mac for the last few days until I've found this guide.
